deep_qa/models/reading_comprehension/bidirectional_attention.py

```diff
@@ -250,6 +250,14 @@ def score_instance(self, instance: CharacterSpanInstance):
 
     @staticmethod
     def get_best_span(span_begin_probs, span_end_probs):
+        if len(span_begin_probs.shape) > 2 or len(span_end_probs.shape) > 2:
```

Comment:
Instead of doing all of this shape inference, why not do `span_begin_probs.squeeze()` and afterwards check it is the right shape. This would deal with `(1, x), (x, 1) and (x)` cases all at once.